src/main/scala/com/microsoft/hyperspace/index/rules/disabled/FilterIndexRule.scala

```diff
@@ -0,0 +1,169 @@
+    if (indexes.isEmpty) {
+      return Map.empty
+    }
+
+    // FilterPlanNodeFilter looks for below patterns, in ordered manner:
+    //  Pattern-1: Scan -> Filter -> Project
+    //  Pattern-2: Scan -> Filter
+    // Pattern-2 covers the case where project node is eliminated or not present.
+    // An example is when all columns are selected.
+    plan match {
+      case Project(_, Filter(_: Expression, ExtractRelation(relation)))
+          if !RuleUtils.isIndexApplied(relation) =>
+        indexes.filterKeys(relation.plan.equals(_))
+      case Filter(_: Expression, ExtractRelation(relation))
+          if !RuleUtils.isIndexApplied(relation) =>
+        indexes.filterKeys(relation.plan.equals(_))
+      case _ =>
+        Map.empty
+    }
+  }
+}
+
+object FilterColumnFilter extends QueryPlanIndexFilter {
+  override def apply(plan: LogicalPlan, indexes: PlanToIndexesMap): PlanToIndexesMap = {
+    if (indexes.isEmpty) {
+      return Map.empty
+    }
+
+    val (filterColumnNames, projectColumnNames) = plan match {
+      case project @ Project(_, _ @Filter(condition: Expression, ExtractRelation(relation)))
+          if !RuleUtils.isIndexApplied(relation) =>
+        val projectColumnNames = CleanupAliases(project)
+          .asInstanceOf[Project]
+          .projectList
+          .map(_.references.map(_.asInstanceOf[AttributeReference].name))
+          .flatMap(_.toSeq)
+        val filterColumnNames = condition.references.map(_.name).toSeq
+        (filterColumnNames, projectColumnNames)
+
+      case Filter(condition: Expression, ExtractRelation(relation))
+          if !RuleUtils.isIndexApplied(relation) =>
+        val relationColumnNames = relation.plan.output.map(_.name)
+        val filterColumnNames = condition.references.map(_.name).toSeq
+        (filterColumnNames, relationColumnNames)
+      case _ =>
+        (Seq(), Seq())
+    }
+
+    // Filter candidate indexes if:
+    //  1. Filter predicate's columns include the first 'indexed' column of the index.
+    //  2. The index covers all columns from the filter predicate and output columns list.
+    val candidateIndexes =
+      indexes.head._2.filter { index =>
+        val ddColumns = index.derivedDataset.properties.columns
+        withReasonTag("The first indexed column should be in filter condition columns.") {
+          ResolverUtils.resolve(spark, ddColumns.indexed.head, filterColumnNames).isDefined
+        }(plan, index) &&
+        withReasonTag(
+          "Index does not contain required columns. Required columns: " +
+            s"[${filterColumnNames ++ projectColumnNames}], indexed & included columns: " +
+            s"[${ddColumns.indexed ++ ddColumns.included}") {
+          ResolverUtils
+            .resolve(
+              spark,
+              filterColumnNames ++ projectColumnNames,
+              ddColumns.indexed ++ ddColumns.included)
+            .isDefined
+        }(plan, index)
+      }
+
+    Map(indexes.head._1 -> candidateIndexes)
```

Comment:
We have `indexes.head._2` above. Can we do:
```scala
// Consider only the first index because ...
val index = indexes.head
```

Reply:
renamed `indexes: PlanToIndexesMap` => `pti: PlanToIndexes`